src/runtimes/docker/utils/normalizePlatform.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerPlatform } from "../contracts/ContainerClient";
+
+/**
+ * Normalizes a platform string or object to a standardized `ContainerPlatform` object.
+ *
+ * @param platform The platform string or object to normalize.
+ * @returns A standardized `ContainerPlatform` object.
+ * @throws An error if the platform string is malformed.
+ */
+export function normalizePlatform(platform: string | ContainerPlatform): ContainerPlatform {
```

Comment:
I would keep normalizePlatform with the task utils, as it is specifically normalizing task specific types to the format expected by ContainerClient and therefore makes sense to belong with the task code.

Reply:
Agreed, especially if it's only used by `src/tasks/DockerBuildTaskProvider.ts`